app/assets/javascripts/react_on_rails.js

```diff
@@ -150,4 +150,15 @@
 
     return consoleReplay;
   }
+
+  function provideProperReact() {
+    if (typeof ReactDOMServer === "undefined") { //not on server
```

Comment:
What if somebody makes a mistake and exposes `ReactDOMServer`, trying to reuse the same Webpack bundle?

I'd rather that we pass a flag from the two places this is called, as we know.

And if possible, check for the positive case of the ReactDOM or ReactDOMServer existing.

```javascript
function provideProperReact(clientRendered) {
} 
```